sdks/cpp/lite/include/Param.h

```diff
@@ -133,6 +138,7 @@ template <typename T> class Param : public IParam {
     std::reference_wrapper<T> value_;
     std::reference_wrapper<Device> dm_;
     std::string widget_;
+    const bool read_only_;
```

Comment:
this shouldn't be a const. it's valid for business logic to change this flag. It's not valid for connection logic to do so though, so IParam should only provide a read accessor for this flag, the write accessor would be provided by either lite::Param or full::Param.